src/chainstate/burn/db/sortdb.rs

```diff
@@ -992,6 +1017,92 @@ impl <'a> SortitionHandleConn <'a> {
         Ok(key_status)
     }
 
+    /// Return a vec of sortition winner's burn header hash and stacks header hash, ordered by
+    ///   increasing block height in the range (block_height_begin, block_height_end]
+    fn get_sortition_winners_in_fork(&self, block_height_begin: u32, block_height_end: u32) -> Result<Vec<(Txid, u64)>,  BurnchainError> {
+        let mut result = vec![];
+        for height in (block_height_begin+1)..(block_height_end+1) {
+            debug!("Looking for winners at height = {}", height);
+            let snapshot = SortitionDB::get_ancestor_snapshot(self, height as u64, &self.context.chain_tip)?
+                .ok_or_else(|| BurnchainError::MissingParentBlock)?;
+            if snapshot.sortition {
+                result.push((snapshot.winning_block_txid, snapshot.block_height));
+            }
+        }
+        Ok(result)
+    }
+
+    pub fn get_reward_cycle_info(&self, prepare_end: &BurnchainHeaderHash, pox_consts: &PoxConstants) -> Result<Option<(ConsensusHash, BlockHeaderHash)>, CoordinatorError> {
+        let prepare_end_sortid = self.get_sortition_id_for_bhh(prepare_end)?
+            .ok_or_else(|| BurnchainError::MissingParentBlock)?;
+        let my_height = SortitionDB::get_block_height(self.deref(), &prepare_end_sortid)?
+            .expect("CORRUPTION: SortitionID known, but no block height in SQL store");
+
+        // there can be either 1 or 0 PoX anchors.
+        assert!(pox_consts.anchor_threshold > (pox_consts.prepare_length / 2));
```

Comment:
Should this check be done as part of the PoxConstants constructor?  Seems odd that we'd do this each time we called this method, when these constants never change.